src/Google.Pubsub.V1/PublisherClient.cs

```diff
@@ -187,10 +159,7 @@ public static class ServiceDefaults
             ServiceEndpointSettings serviceEndpointSettings = null)
         {
             serviceEndpointSettings = serviceEndpointSettings ?? GetDefaultServiceEndpointSettings();
-            Channel channel = new Channel(
-                serviceEndpointSettings.Host ?? ServiceDefaults.Host,
-                serviceEndpointSettings.Port ?? ServiceDefaults.Port,
-                credentials);
+            Channel channel = ClientHelper.CreateChannel(serviceEndpointSettings, ServiceDefaults.Host, ServiceDefaults.Port, credentials);
```

Comment:
I would prefer removing the mutation to serviceEndpointSettings:
Channel channel = ClientHelper.CreateChannel(serviceEndpointSettings ?? GetDefaultServiceEndpointSettings(),
    ServiceDefaults.Host, ServiceDefault.Port, credentials);
(and in SubscriberClient)